istioctl/cmd/vm-bootstrap.go

```diff
@@ -0,0 +1,630 @@
+				if err != nil {
+					return err
+				}
+				decryptedKey, err := ssh.ParsePrivateKeyWithPassphrase(rawKey, []byte(sshKeyPassword))
+				if err != nil {
+					return err
+				}
+				sshAuthMethod = ssh.PublicKeys(decryptedKey)
+			} else {
+				return err
+			}
+		} else {
+			sshAuthMethod = ssh.PublicKeys(key)
+		}
+	}
+
+	return nil
+}
+
+func vmBootstrapCommand() *cobra.Command {
+	vmBSCommand := &cobra.Command{
+		Use:   "vm-bootstrap",
+		Short: "bootstraps a workloadentry representing a vm into the mesh",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if (len(args) == 1) == all {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("vm-bootstrap requires a workload entry, or the all flag")
+			}
+			if all && workloadNamespace == "" {
+				return fmt.Errorf("vm-bootstrap needs a namespace if fetching all workspaces")
+			}
+			if sshUser == "" {
+				user, err := user.Current()
+				if err != nil {
+					return err
+				}
+				sshUser = user.Username
+			}
+			if dumpDir == "" {
+				err := deriveSSHMethod()
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			var configClient model.ConfigStore
+			var err error
+
+			if configClient, err = clientFactory(); err != nil {
+				return err
+			}
+
+			var entries []model.Config
+			var chosenNS string
+			if all {
+				entries, chosenNS, err = fetchAllWorkloadEntries(configClient)
+			} else {
+				entries, chosenNS, err = fetchSingleWorkloadEntry(args[0], configClient)
+			}
+			if err != nil {
+				return err
+			}
+
+			kubeClient, err := interfaceFactory(kubeconfig)
+			if err != nil {
+				return err
+			}
+
+			certs, err := getCertificate(kubeClient)
+			if err != nil {
+				return err
+			}
+
+			addresses, err := getCertificatesForEachAddress(entries, chosenNS, certs)
+			if err != nil {
+				return err
+			}
+
+			if dumpDir != "" {
+				err = dumpCertificates(dumpDir, addresses)
+				if err != nil {
+					return err
+				}
+			} else {
+				err = copyCertificates(kubeClient, addresses)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+
+	vmBSCommand.PersistentFlags().BoolVarP(&all, "all", "a", false,
+		"attempt to bootstrap all workload entries")
+	vmBSCommand.PersistentFlags().StringVarP(&dumpDir, "dump-directory", "d", "",
+		"dump the certificates locally in the provided directory instead of uploading remotely.")
+	vmBSCommand.PersistentFlags().DurationVar(&certDuration, "duration", 365*24*time.Hour,
+		"duration the certificates generated are valid for.")
+	vmBSCommand.PersistentFlags().StringVar(&istioProxyImage, "istio-image", "istio/proxyv2:latest",
+		"the istio proxy image to start up when starting istio")
+	vmBSCommand.PersistentFlags().BoolVar(&mutualTLS, "mutual-tls", false,
+		"Whether or not to enable mutual TLS if starting istio-proxy.")
+	vmBSCommand.PersistentFlags().StringVarP(&workloadNamespace, "namespace", "n", "",
```

Comment:
There is already a namespace flag in _istioctl/cmd/root.go_ that will be inherited.  Can you use that instead?

Reply:
Yep, happy to inherit where possible.